request_changes.py

```diff
@@ -0,0 +1,371 @@
+import json
+import logging
+import os
+import re
+import time
+from datetime import datetime
+
+import psycopg2
+import requests
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+start_time = time.time()
+
+logging.info("-------------------------REQUEST CHANGES SCRIPT IS RUNNING-------------------------")
+
+gitea_api_endpoint = "https://gitea.eco.tsi-dev.otc-service.com/api/v1"
+session = requests.Session()
+gitea_token = os.getenv("GITEA_TOKEN")
+
+db_host = os.getenv("DB_HOST")
+db_port = os.getenv("DB_PORT")
+db_csv = os.getenv("DB_CSV")  # main postgres db, where open PRs tables for both public and hybrid clouds are stored
+db_user = os.getenv("DB_USER")
+db_password = os.getenv("DB_PASSWORD")
+
+
+def check_env_variables():
+    required_env_vars = [
+        "DB_HOST", "DB_PORT",
+        "DB_CSV", "DB_USER", "DB_PASSWORD", "GITEA_TOKEN"
+    ]
+    for var in required_env_vars:
+        if os.getenv(var) is None:
+            raise Exception("Missing environment variable: %s", var)
+
+
+def connect_to_db(db_name):
+    logging.info("Connecting to Postgres (%s)...", db_name)
+    try:
+        return psycopg2.connect(
+            host=db_host,
+            port=db_port,
+            dbname=db_name,
+            user=db_user,
+            password=db_password
+        )
+    except psycopg2.Error as e:
+        logging.error("Connecting to Postgres: an error occurred while trying to connect to the database: %s", e)
+        return None
```

Comment:
i don't understand why are you using same code in all files, please introduce db class and reuse it

Reply:
same for env variables